ServerCore/Pages/Puzzles/FileManagement.cshtml.cs

```diff
@@ -287,6 +287,36 @@ private bool PuzzleExists(int puzzleId)
             return _context.Puzzles.Any(e => e.ID == puzzleId);
         }
 
+        /// <summary>
+        /// Promotes files named "index.html" to the CustomURL or CustomSolutionURL properties of the puzzle.
+        /// </summary>
+        /// <param name="file">The file being uploaded</param>
+        private void UpdateCustomURLs(ContentFile file)
+        {
+            // Skip this if:
+            // - the file is not named index.html
+            // - the file is not a material or solution file
+            // - the appropriate Custom[Solution]URL is non-blank
+            if (!file.ShortName.EndsWith("index.html") ||
```

Comment:
[nit]: These are a lot of "!" which can be a little hard to read.

Perhaps call it
bool shouldUpdateCustomUrl = file.ShortName.EndsWith(index.html) && (file.FileType == solution || fileType == Solve) && etc.

if (!shouldUpdateCustomUrl){ return}

Reply:
I don't feel that the alternate version is simpler in this case.